x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -664,6 +678,37 @@ private Join resolveLookupJoin(LookupJoin join) {
             return join;
         }
 
+        private LogicalPlan resolveFork(Fork fork, AnalyzerContext context) {
+            List<LogicalPlan> subPlans = fork.subPlans();
+            NameId firstForkNameId = null;  // stores the id of the first _fork,
+
+            List<LogicalPlan> newSubPlans = new ArrayList<>();
+            for (var logicalPlan : subPlans) {
+                LogicalPlan newPlan = logicalPlan.transformUp(LogicalPlan.class, p -> p.childrenResolved() == false ? p : rule(p, context));
+
+                // align _fork id across all fork branches
+                Eval eval = forkEvalOrThrow(newPlan);
+                if (firstForkNameId == null) {
+                    firstForkNameId = eval.fields().getFirst().id();
+                } else {
+                    var literal = eval.fields().getFirst().child();
+                    var alias = new Alias(eval.source(), "_fork", literal, firstForkNameId);
+                    newPlan = new Eval(eval.source(), eval.child(), List.of(alias));
+                }
+                newSubPlans.add(newPlan);
+            }
```

Comment:
Move this logic to the parser (visitForkCommand), similar to  the approach done visitForkSubQueries. You already know the amount of subqueries you have so it's easy to pick the first one.